### ISSUE TITLE
[QUERY] How to access existing long running operation later

### ISSUE THREAD

**Issue by ranul-pallemulle** [Azure/azure-sdk-for-net#27409]
### Library name and version

Azure.ResourceManager.Resources

### Query/Question

I am using the latest available beta version of the library at this time (1.0.0-beta.6).
I am creating a new resource group as follows:
```
var rgCollection = subscription.GetResourceGroups();
var operation = await rgCollection.CreateOrUdpateAsync(false, "MyResourceGroup", new ResourceGroupData(AzureLocation.WestUS2));
var myOperationId = operation.Id;
```

According to the guidelines here:
https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-longrunning
'Client libraries need to inherit from Operation<T> not only to implement all abstract members, but also to provide a constructor required to access an existing LRO (an LRO initiated by a different process).'
Is this rule applicable to management libraries such as this one as well?

Basically what I want to do is use the operation ID to get access to a ResourceGroupCreateOrUpdateOperation object. Something like this:
```
var operation = new ResourceGroupCreateOrUpdateOperation(myOperationId);
await operation.WaitForCompletionAsync();
```
The reason I need this is that my application server may encounter some problem and restart during the long running operation. I need some way to check later if some long running operation I've initiated has completed.
Is there a way to do this at the moment? 

### Environment

OS: Mac OS X
.NET runtime version: Microsoft.NETCore.App 6.0.0

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by ArthurMa1978** [comment]
@ranul-pallemulle good question, we are working on redesigning the LRO, since it has not been finalized so the doc https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-longrunning has not been updated.
The beta sdk you are using will also be changed.
I'll get back to you once it's done.

**Comment by ArthurMa1978** [comment]
We have another https://github.com/Azure/autorest.csharp/issues/1622 opened for tracking this functionality
